src/mlpack/tests/nmf_test.cpp

```diff
@@ -222,4 +222,66 @@ BOOST_AUTO_TEST_CASE(SparseNMFALSTest)
       1e-5);
 }
 
-BOOST_AUTO_TEST_SUITE_END();
+/**
+ * Check if all elements in W and H are non-negative.
+ * Default Case.
+ * Random Acol initilization and distance minimization update.
+ */
+BOOST_AUTO_TEST_CASE(NonNegNMFDefaultTest)
+{
+  mat w = randu<mat>(20, 12);
+  mat h = randu<mat>(12, 20);
+  mat v = w * h;
+  const size_t r = 12;
+
+  AMF<> nmf;
+  nmf.Apply(v, r, w, h);
+
+  BOOST_REQUIRE(arma::all(vectorise(w) >= 0)
+      && arma::all(vectorise(h) >= 0));
+}
+
+/**
+ * Check if all elements in W and H are non-negative.
+ * Random initialization divergence minimization update.
+ */
+BOOST_AUTO_TEST_CASE(NonNegNMFRandomDivTest)
+{
+  mat w = randu<mat>(20, 12);
+  mat h = randu<mat>(12, 20);
+  mat v = w * h;
+  const size_t r = 12;
+
+  // Custom tighter tolerance.
+  SimpleResidueTermination srt(1e-8, 10000);
+  AMF<SimpleResidueTermination,
+      RandomInitialization,
```

Comment:
Any reason for not using `RandomAcolInitialization` here?

Reply:
Because in nmf_main.cpp, the factorizer for `updateRules == "multdiv"` uses `RandomInitialization`.

Reply:
Ok, thanks for the clarification.  I should have looked in `nmf_main.cpp` to answer my own question, sorry about that.

Reply:
don't worry, I'm happy to help!